### PR TITLE
Added new functionality in scope of perf-427

### PR DESCRIPTION
**Pull request by Roman-Fedynyshyn**

### CODE REVIEW

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                "</style>\n" +
+                "<table class=\"tg\">\n" +
+                "    <thead>\n" +
+                "    <tr>\n" +
+                "        <th class=\"tg-73oq\">Module<br>"+cluster+"<br>  "+date+"</th>\n" +
+                "        <th class=\"tg-73oq\">Revision</th>\n" +
+                "        <th class=\"tg-73oq\">Version</th>\n" +
+                "        <th class=\"tg-73oq\">Count</th>\n" +
+                "        <th class=\"tg-73oq\"><span style=\"font-weight:400;font-style:normal\">Hard LImit</span></th>\n" +
+                "        <th class=\"tg-73oq\"><span style=\"font-weight:400;font-style:normal\">Soft limit</span></th>\n" +
```
Comment (mtraneis) [inline common/ECSResourcesCheck/src/main/java/Writer.java:28]: May as well throw this style into a class and refer to it like you are doing elsewhere.

File: common/ECSResourcesCheck/README.md [hunk 2]
```diff
@@ -3,16 +3,20 @@
...
+     
+     aws_access_key_id,
+     
+     aws_secret_access_key,
+     
+     aws region,
+     
+     behavior - get - to get description of services/ null - to compare with base json,  
+                    
+     modules list - coma separated values of modules names,
```
Comment (mtraneis) [inline common/ECSResourcesCheck/README.md:16]: Please specify when the modules list or the base json field can be optional or required.

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Date;
+
+public class Writer {
+    public void write(JSONObject json, Date date, String cluster)
```
[the remark this answers is not in the document]
Comment (mtraneis) [reply common/ECSResourcesCheck/src/main/java/Writer.java:10]: Should check if "json" is null

**Review by mtraneis — COMMENTED** [review]
If you want to tidy up the main function, each of the code branch under _if_ or _else_ can be factored out into a function so that all you have to do under if/else is to call the function.  You also printed out "I'm retrieving.." or "it's getting cluster info", etc.. which implies that these are distinct actions that can be in their own functions.

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                "  font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;padding:10px 3px;word-break:normal;}\n" +
+                ".tg .tg-73oq{border-color:#000000;text-align:left;vertical-align:top}\n" +
+                "</style>\n" +
+                "<table class=\"tg\">\n" +
+                "    <thead>\n" +
+                "    <tr>\n" +
+                "        <th class=\"tg-73oq\">Module<br>"+cluster+"<br>  "+date+"</th>\n" +
+                "        <th class=\"tg-73oq\">Revision</th>\n" +
+                "        <th class=\"tg-73oq\">Version</th>\n" +
+                "        <th class=\"tg-73oq\">Count</th>\n" +
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:26]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                ".tg th{background-color:#f0f0f0;border-color:#ccc;border-style:solid;border-width:0px;color:#333;\n" +
+                "  font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;padding:10px 3px;word-break:normal;}\n" +
+                ".tg .tg-73oq{border-color:#000000;text-align:left;vertical-align:top}\n" +
+                "</style>\n" +
+                "<table class=\"tg\">\n" +
+                "    <thead>\n" +
+                "    <tr>\n" +
+                "        <th class=\"tg-73oq\">Module<br>"+cluster+"<br>  "+date+"</th>\n" +
+                "        <th class=\"tg-73oq\">Revision</th>\n" +
+                "        <th class=\"tg-73oq\">Version</th>\n" +
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:25]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                "  font-family:Arial, sans-serif;font-size:14px;overflow:hidden;padding:10px 3px;word-break:normal;}\n" +
+                ".tg th{background-color:#f0f0f0;border-color:#ccc;border-style:solid;border-width:0px;color:#333;\n" +
+                "  font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;padding:10px 3px;word-break:normal;}\n" +
+                ".tg .tg-73oq{border-color:#000000;text-align:left;vertical-align:top}\n" +
+                "</style>\n" +
+                "<table class=\"tg\">\n" +
+                "    <thead>\n" +
+                "    <tr>\n" +
+                "        <th class=\"tg-73oq\">Module<br>"+cluster+"<br>  "+date+"</th>\n" +
+                "        <th class=\"tg-73oq\">Revision</th>\n" +
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:24]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                "        <th class=\"tg-73oq\">Version</th>\n" +
+                "        <th class=\"tg-73oq\">Count</th>\n" +
+                "        <th class=\"tg-73oq\"><span style=\"font-weight:400;font-style:normal\">Hard LImit</span></th>\n" +
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:27]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:28]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                "        <th class=\"tg-73oq\"><span style=\"font-weight:400;font-style:normal\">Soft limit</span></th>\n" +
+                "        <th class=\"tg-73oq\">CPU units</th>\n" +
+                "        <th class=\"tg-73oq\">xmx</th>\n" +
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:30]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+                "        <th class=\"tg-73oq\">metaspace</th>\n" +
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:31]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+                "        <th class=\"tg-73oq\">maxMetaspace</th>\n" +
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:32]: done

File: common/ECSResourcesCheck/src/main/java/Main.java [hunk 11]
```diff
@@ -1,16 +1,58 @@
...
+                    args[1], //aws_access_key_id,
+                    args[2], //aws_secret_access_key,
+                    args[3]);  //aws region
+
+            if (args[5].equals("null")) {
+                System.out.println("it's getting cluster info");
+                json = ecsServices.getServiceslist();
+                writer.write(json, date, args[0]);
+            }
+
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Main.java:38]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+                "        <th class=\"tg-73oq\">R/W split enabled</th>\n" +
+                "    </tr>\n" +
+                "    </thead>\n" +
+                "    <tbody>";
+        BufferedWriter writer = new BufferedWriter(new FileWriter("clusterResources.html"));
+        writer.write(Head);
+        for (int i = 0; i < json.length(); i++) {
+            JSONArray a=json.names();
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:40]: renamed to "names"

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Date;
+
+public class Writer {
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:9]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+                "        <th class=\"tg-73oq\">R/W split enabled</th>\n" +
+                "    </tr>\n" +
+                "    </thead>\n" +
+                "    <tbody>";
+        BufferedWriter writer = new BufferedWriter(new FileWriter("clusterResources.html"));
+        writer.write(Head);
+        for (int i = 0; i < json.length(); i++) {
+            JSONArray a=json.names();
+
+            JSONObject aa= json.getJSONObject(a.get(i).toString());//really?
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:42]: > Rename "aa" to something more meaningful and remove "//really?"

done

File: common/ECSResourcesCheck/src/main/java/Writer.java [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+                    "        <td class=\"tg-73oq\">"+aa.getInt("XMX")+"</td>\n" +
+                    "        <td class=\"tg-73oq\">"+aa.getInt("Metaspace")+"</td>\n" +
+                    "        <td class=\"tg-73oq\">"+aa.getInt("MaxMetaspaceSize")+"</td>\n" +
+                    "        <td class=\"tg-73oq\">"+aa.getBoolean("RWSplitEnabled")+" </td>\n" +
+                    "    </tr>";
+            writer.append(s);
+        }
+        String footer =" </tbody>\n" +
+                "</table>";
+
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java:61]: done

File: common/ECSResourcesCheck/src/main/java/Main.java [hunk 16]
```diff
@@ -1,16 +1,58 @@
...
+        if (args[4].equals("get")) {
+            Writer writer = new Writer();
+            JSONObject json = new JSONObject();
+            ECSServices ecsServices = new ECSServices(
+                    args[0], //ecs cluster name
+                    args[1], //aws_access_key_id,
+                    args[2], //aws_secret_access_key,
+                    args[3]);  //aws region
+
```
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Main.java:32]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java]: done

File: common/ECSResourcesCheck/src/main/java/Writer.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (Roman-Fedynyshyn) [reply common/ECSResourcesCheck/src/main/java/Writer.java]: added functionality to break a loop in case if json is null.

**Pull request merged**